### ISSUE TITLE
Instructions to rebuild?

### ISSUE THREAD

**Issue by ALRubinger** [TBD54566975/ssi-service#72]
More newb questions I'm hoping help new folks get going.

In #68, @frankhinek noted to me:

> make sure you rebuild the build_web_1 container after pulling in the latest commit(s):

```
cd build
docker-compose build && docker-compose up 
```

I was expecting from the `README` that `mage cbt` and (`mage run` | `cd build && docker-compose up`) would accomplish that for me, and found that I wasn't getting new changes picked up in the deployment. The recommendation above did the trick.

Is this accurate, and should we update either the docs to show this or the `cbt` step to fully rebuild things?

**Comment by frankhinek** [comment]
Docker Compose only recreates containers if their configuration files have changed, and by default, preserves all volumes used by services.  When developing locally, we'll regularly need to rebuild container images to pickup source code changes.

If the intent is for local development environments to use Docker containers to ensure consistent builds, you could add commands to the `magefile` to address cached container artifacts.  Since the `Run()` function uses `docker-compose`, I assume that's the case.

You could add a Docker Compose `build` step to CBT to force the image rebuild:

```go
sh.Run("docker-compose", "--project-directory", "build", "build")
```

Alternatively (or additionally), you can add clean function to remove the locally cached container images.  The next time `docker-compose up` is run, the images would have to be rebuilt.

```go
sh.Run("docker-compose", "--project-directory", "build", "down", "--rmi", "local")
```

**Comment by decentralgabe** [comment]
Thanks @frankhinek this is great. Would you be willing to update the mage file? or perhaps a "mage cleanRun" command that facilitates what you've described?

I've assigned the issue to you for now. If you don't want to do it, feel free to let me know 😄

**Comment by ALRubinger** [comment]
Appreciate you, @frankhinek! Smoothing out these build intricacies really make the difference to newcomers and give them a joyful, predictable experience.

**Comment by frankhinek** [comment]
PR https://github.com/TBD54566975/ssi-service/pull/77 opened.

**Comment by decentralgabe** [comment]
Fixed in #77 thanks @frankhinek !
